src/Connector.SqlServer/Utils/SqlName.cs

```diff
@@ -17,15 +17,15 @@ public static SqlName FromUnsafe(string value)
             if (string.IsNullOrEmpty(value)) throw new ArgumentException("Value cannot be null or empty.", nameof(value));
 
             var sanitizedValue = value.ToSanitizedSqlName();
-            if (string.IsNullOrEmpty(sanitizedValue))
+            if (string.IsNullOrEmpty(sanitizedValue)) 
```

Comment:
@zvirja 
From one perspective this looks as extra check and could be skipped if we route our execution from this point to `FromSanitized` .